modules/xfem/src/base/XFEM.C

```diff
@@ -1733,41 +1695,59 @@ XFEM::setSolution(SystemBase & sys,
                   NumericVector<Number> & old_solution,
                   NumericVector<Number> & older_solution)
 {
-  for (auto dit = stored_solution.begin(); dit != stored_solution.end(); ++dit)
+  for (auto node_it = _mesh->local_nodes_begin(); node_it != _mesh->nodes_end(); ++node_it)
```

Comment:
I'd move the `nodes_end()` call out of the loop. The libMesh iterators can be slow to evaluate.